src/engine/OptionalJoin.cpp

```diff
@@ -66,6 +68,9 @@ OptionalJoin::OptionalJoin(QueryExecutionContext* qec,
 string OptionalJoin::getCacheKeyImpl() const {
   std::ostringstream os;
   os << "OPTIONAL_JOIN\n" << _left->getCacheKey() << " ";
+  if (!keepJoinColumns_) {
+    os << "Dropping join-columns";
+  }
```

Comment:
This works, but I believe this part of the cache key should not come in-between the key of the right and the left child and in the front/end instead.